src/core/mac/mac_frame.cpp

```diff
@@ -1518,6 +1623,62 @@ Error TxFrame::GenerateEnhAck(const RxFrame &aRxFrame, bool aIsFramePending, con
 }
 #endif // OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2
 
+#if OPENTHREAD_CONFIG_WAKEUP_COORDINATOR_ENABLE
+Error TxFrame::GenerateWakeupFrame(PanId aPanId, const Address &aDest, const Address &aSource)
+{
+    Error    error = kErrorNone;
+    uint16_t fcf   = kTypeMultipurpose | kMpFcfLongFrame | kMpFcfPanidPresent | kMpFcfSecurityEnabled |
+                   kMpFcfSequenceSuppression | kMpFcfIePresent;
+    uint8_t index = 0;
+
+    switch (aDest.GetType())
+    {
+    case Address::Type::kTypeShort:
+        fcf |= kMpFcfDstAddrShort;
+        break;
+    case Address::Type::kTypeExtended:
+        fcf |= kMpFcfDstAddrExt;
+        break;
+    default:
+        ExitNow(error = kErrorInvalidArgs);
+        break;
+    }
+
+    switch (aSource.GetType())
+    {
+    case Address::Type::kTypeShort:
+        fcf |= kMpFcfSrcAddrShort;
+        break;
+    case Address::Type::kTypeExtended:
+        fcf |= kMpFcfSrcAddrExt;
+        break;
+    default:
+        ExitNow(error = kErrorInvalidArgs);
+        break;
+    }
+
+    mLength = CalculateAddrFieldSize(fcf);
+
+    OT_ASSERT(mLength != kInvalidSize);
+
+    SetFrameControlField(fcf);
+    SetDstPanId(aPanId);
+    SetDstAddr(aDest);
+    SetSrcAddr(aSource);
+
+    mPsdu[mLength] = Mac::Frame::kKeyIdMode2 | Mac::Frame::kSecurityEncMic32;
```

Comment:
Can remove `Mac::Frame::` here and below?

Reply:
Done